ccd-claim-migration/src/main/java/uk/gov/hmcts/cmc/ccd/migration/ccd/services/MigrateCoreCaseDataService.java

```diff
@@ -0,0 +1,179 @@
+public class MigrateCoreCaseDataService {
+
+    private static final Logger logger = LoggerFactory.getLogger(MigrateCoreCaseDataService.class);
+
+    private final CoreCaseDataApi coreCaseDataApi;
+    private final AuthTokenGenerator authTokenGenerator;
+    private final CaseAccessApi caseAccessApi;
+    private final UserService userService;
+    private final CaseMapper caseMapper;
+
+    @Autowired
+    public MigrateCoreCaseDataService(
+        CoreCaseDataApi coreCaseDataApi,
+        AuthTokenGenerator authTokenGenerator,
+        CaseAccessApi caseAccessApi,
+        UserService userService,
+        CaseMapper caseMapper
+    ) {
+        this.coreCaseDataApi = coreCaseDataApi;
+        this.authTokenGenerator = authTokenGenerator;
+        this.caseAccessApi = caseAccessApi;
+        this.userService = userService;
+        this.caseMapper = caseMapper;
+    }
+
+    public void update(
+        String authorisation,
+        EventRequestData eventRequestData,
+        Long ccdId,
+        Claim claim
+    ) {
+        logger.info("claim: " + claim);
+        CCDCase ccdCase = caseMapper.to(claim);
+        logger.info("ccdCase: " + ccdCase);
+        StartEventResponse startEventResponse = startEvent(authorisation, eventRequestData, ccdId);
+
+        CaseDataContent caseDataContent = CaseDataContent.builder()
+            .eventToken(startEventResponse.getToken())
+            .event(
+                Event.builder()
+                    .id(startEventResponse.getEventId())
+                    .summary("CMC case migrate")
+                    .description("Submitting CMC case migrate - overwrite")
+                    .build()
+            ).data(ccdCase)
+            .build();
+
+        submitEvent(authorisation, eventRequestData, caseDataContent, ccdId);
+    }
+
+    public void save(
+        String authorisation, EventRequestData eventRequestData, Claim claim
+    ) {
+        logger.info("claim: " + claim);
+        CCDCase ccdCase = caseMapper.to(claim);
+        logger.info("ccdCase: " + ccdCase);
+
+        StartEventResponse startEventResponse = start(authorisation, eventRequestData);
+
+        CaseDataContent caseDataContent = CaseDataContent.builder()
+            .eventToken(startEventResponse.getToken())
+            .event(
+                Event.builder()
+                    .id(startEventResponse.getEventId())
+                    .summary("CMC case migrate")
+                    .description("Submitting CMC case migrate - create")
+                    .build()
+            ).data(ccdCase)
+            .build();
+
+        CaseDetails caseDetails = submit(authorisation, eventRequestData, caseDataContent);
+
+        String defendantId = claim.getDefendantId() != null ? claim.getDefendantId() : claim.getLetterHolderId();
+
+        // make sure both submitter and defendant (or letterHolder) can access the case
+        grantAccess(caseDetails.getId().toString(), claim.getSubmitterId());
```

Comment:
This probably needs to be run on update/overwrite as well.

Reply:
yes, updated